src/Hint/Lambda.hs

```diff
@@ -268,7 +274,31 @@ fromLambda (SimpleLambda ps1 (fromLambda . fromParen -> (ps2,x))) = (transformBi
           f bad x = x
 fromLambda x = ([], x)
 
--- | Replaces all non-wildcard patterns with a variable pattern with the given identifier.
-munge :: Char -> LPat GhcPs -> LPat GhcPs
-munge ident p@(L _ (WildPat _)) = p
-munge ident (L ploc p) = L ploc (VarPat noExtField (L ploc $ mkRdrUnqual $ mkVarOcc [ident]))
+-- | For each pattern, if it does not contain wildcards, replace it with a variable pattern.
+--
+-- The second component of the result is a list of substitution variables, which is ['a'..'z'],
+-- excluding variables that occur in patterns with wildcards. For example, if there is a pattern
+-- 'Foo a b _', then 'a' and 'b' are removed.
+mkOrigPats :: [LPat GhcPs] -> ([LPat GhcPs], String)
```

Comment:
Better to use `[Char]` when that is the moral underlying type, rather than String.

```suggestion
mkOrigPats :: [LPat GhcPs] -> ([LPat GhcPs], [Char])
```

Reply:
I thought this would cause the CI to fail since it triggers the "Use String" hint - I'll see if CI cares or not.

Reply:
It does - I disabled it in .hlint.yaml.